discovery-infra/tests/base_test.py

```diff
@@ -26,20 +36,42 @@ def node_controller(self):
         controller.shutdown_all_nodes()
         controller.format_all_node_disks()
 
+    @pytest.fixture()
+    def cluster(self, api_client):
+        clusters = []
+
+        def get_cluster_func(cluster_name: Optional[str] = None):
+            if not cluster_name:
+                cluster_name = random_name()
+
+            res = api_client.create_cluster(cluster_name,
+                                            ssh_public_key=env_variables['SSH_PUBLIC_KEY'],
+                                            openshift_version=env_variables['OPENSHIFT_VERSION'],
+                                            pull_secret=env_variables['PULL_SECRET'],
+                                            base_dns_domain=env_variables['BASE_DOMAIN'],
+                                            vip_dhcp_allocation=env_variables['VIP_DHCP_ALLOCATION'])
+            clusters.append(res)
+            return res
+
+        yield get_cluster_func
+
+        for cluster in clusters:
+            with suppress(ApiException):
+                api_client.delete_cluster(cluster.id)
+
+    @pytest.fixture()
```

Comment:
This can only safely be used on a local minikube deployment
Maybe we should not have it at all. The cluster fixture you wrote all ready cleans up the clusters created durrng the test